### PR TITLE
Add "social sharing image" image chooser to settings panel

### PR DESCRIPTION
**Pull request by contolini**
This adds a new image chooser field that populates the HTML Open Graph meta tags (which is what Facebook and Twitter use when you share a page).

The logic in our base template has been changed to:

```
IF page.social_image USE page.social_image
OTHERWISE IF page.preview_image USE page.preview_image
OTHERWISE USE default_cfpb_logo
```

I also added some logic to instruct Twitter to use large cards if the supplied social sharing image is large enough.

## Additions

- Social image chooser in the settings panel of page's admin screen
- Large twitter card option

## Changes

- `meta` image tag logic

### New social sharing image field in the settings panel:

screen shot 2017-05-30 at 5 46 25 pm

### Current normal size Twitter cards:

screen shot 2017-05-30 at 12 49 56 pm

### New large Twitter cards:

screen shot 2017-05-30 at 12 56 06 pm

## Notes

- I added the field to the `CFGOVPage` class which means almost every page on the website will have it available. @ielerol @schaferjh Let me know if you think we should instead restrict it to certain page types (blog post, askcfpb pages, etc.).

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by schaferjh** [comment]
👍  to including it on all page types

**Comment by richaagarwal** [comment]
Could we get a couple unit tests please :)

**Comment by Scotchester** [comment]
> @contolini pushed 156 commits.

Looks like we got a rebasing problem :(

File: cfgov/v1/tests/models/test_base.py [hunk 1]
```diff
@@ -258,3 +260,21 @@ def test_assemble_csv(self):
...
+        self.social_sharing_image = mommy.prepare(CFGOVImage)
+
+    def test_meta_image_no_images(self):
+        page = mommy.prepare(
+            CFGOVPage,
+            social_sharing_image=None
+        )
+        self.assertIsNone(page.meta_image)
+
+    def test_meta_image(self):
```
Comment (richaagarwal) [inline cfgov/v1/tests/models/test_base.py]: Could you describe what this test does, either in the name, or in a docstring?  It's unclear what behavior it is verifying.

**Pull request merged**